profitbricks/src/test/java/org/jclouds/profitbricks/features/ImageApiMockTest.java

```diff
@@ -0,0 +1,114 @@
+ * limitations under the License.
+ */
+package org.jclouds.profitbricks.features;
+
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+import java.util.List;
+import org.jclouds.profitbricks.ProfitBricksApi;
+import org.jclouds.profitbricks.domain.Image;
+import org.jclouds.profitbricks.internal.BaseProfitBricksMockTest;
+import static org.jclouds.profitbricks.internal.BaseProfitBricksMockTest.mockWebServer;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+import org.testng.annotations.Test;
+
+/**
+ * Mock tests for the {@link org.jclouds.profitbricks.features.ImageApi} class
+ */
+@Test(groups = "unit", testName = "ImageApiMockTest")
+public class ImageApiMockTest extends BaseProfitBricksMockTest {
+
+   @Test
+   public void testGetAllImages() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setBody(payloadFromResource("/image/images.xml")));
+
+      ProfitBricksApi pbApi = api(server.getUrl(rootUrl));
+      ImageApi api = pbApi.imageApi();
+
+      try {
+         List<Image> images = api.getAllImages();
+         assertRequestHasCommonProperties(server.takeRequest());
+         assertNotNull(images);
+         assertTrue(images.size() == 7);
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+   }
+
+   @Test
+   public void testGetAllImagesReturning404() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setResponseCode(404));
+
+      ProfitBricksApi pbApi = api(server.getUrl(rootUrl));
+      ImageApi api = pbApi.imageApi();
+
+      try {
+         List<Image> images = api.getAllImages();
+         assertRequestHasCommonProperties(server.takeRequest());
+         assertTrue(images.isEmpty());
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+   }
+
+   @Test
+   public void testGetImage() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setBody(payloadFromResource("/image/image.xml")));
+
+      ProfitBricksApi pbApi = api(server.getUrl(rootUrl));
+      ImageApi api = pbApi.imageApi();
+
+      String id = "5ad99c9e-9166-11e4-9d74-52540066fee9";
+      try {
+         Image image = api.getImage(id);
+         assertRequestHasCommonProperties(server.takeRequest());
+         assertNotNull(image);
+         assertEquals(image.id(), id);
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+
+   }
+
+   @Test
+   public void testGetNonExistingImage() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setResponseCode(500).setBody(payloadFromResource("/image/image-not-found.xml")));
```

Comment:
Hmmm this test depends on the `ResponseStatusFromPayloadHttpCommandExecutorService` fixing the response code. Even if the *real* provider returns a 500, I'd just configure a 404 response to make sure the API works, without coupling it to that class.
The `ResponseStatusFromPayloadHttpCommandExecutorService` tests should take care the status is properly fixed and that should be enough.